<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
 "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Laurie Stevison" />
  <title>R on the Command Line</title>
  <style type="text/css">
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
            pre > code.sourceCode { white-space: pre; position: relative; }
            pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
            pre > code.sourceCode > span:empty { height: 1.2em; }
            .sourceCode { overflow: visible; }
            code.sourceCode > span { color: inherit; text-decoration: inherit; }
            div.sourceCode { margin: 1em 0; }
            pre.sourceCode { margin: 0; }
            @media screen {
            div.sourceCode { overflow: auto; }
            }
            @media print {
            pre > code.sourceCode { white-space: pre-wrap; }
            pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
            }
            pre.numberSource code
              { counter-reset: source-line 0; }
            pre.numberSource code > span
              { position: relative; left: -4em; counter-increment: source-line; }
            pre.numberSource code > span > a:first-child::before
              { content: counter(source-line);
                position: relative; left: -1em; text-align: right; vertical-align: baseline;
                border: none; display: inline-block;
                -webkit-touch-callout: none; -webkit-user-select: none;
                -khtml-user-select: none; -moz-user-select: none;
                -ms-user-select: none; user-select: none;
                padding: 0 4px; width: 4em;
                color: #aaaaaa;
              }
            pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
            div.sourceCode
              {   }
            @media screen {
            pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
            }
            code span.al { color: #ff0000; font-weight: bold; } /* Alert */
            code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
            code span.at { color: #7d9029; } /* Attribute */
            code span.bn { color: #40a070; } /* BaseN */
            code span.bu { color: #008000; } /* BuiltIn */
            code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
            code span.ch { color: #4070a0; } /* Char */
            code span.cn { color: #880000; } /* Constant */
            code span.co { color: #60a0b0; font-style: italic; } /* Comment */
            code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
            code span.do { color: #ba2121; font-style: italic; } /* Documentation */
            code span.dt { color: #902000; } /* DataType */
            code span.dv { color: #40a070; } /* DecVal */
            code span.er { color: #ff0000; font-weight: bold; } /* Error */
            code span.ex { } /* Extension */
            code span.fl { color: #40a070; } /* Float */
            code span.fu { color: #06287e; } /* Function */
            code span.im { color: #008000; font-weight: bold; } /* Import */
            code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
            code span.kw { color: #007020; font-weight: bold; } /* Keyword */
            code span.op { color: #666666; } /* Operator */
            code span.ot { color: #007020; } /* Other */
            code span.pp { color: #bc7a00; } /* Preprocessor */
            code span.sc { color: #4070a0; } /* SpecialChar */
            code span.ss { color: #bb6688; } /* SpecialString */
            code span.st { color: #4070a0; } /* String */
            code span.va { color: #19177c; } /* Variable */
            code span.vs { color: #4070a0; } /* VerbatimString */
            code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
          </style>
  <style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
  </style>
  <script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
  <link href="site_libs/slidy-2/styles/slidy.css" rel="stylesheet" />
  <script src="site_libs/slidy-2/scripts/slidy.js"></script>
  <script src="site_libs/slidy_shiny-1/slidy_shiny.js"></script>
</head>
<body>
<div class="slide titlepage">
  <h1 class="title">R on the Command Line</h1>
  <p class="author">
Laurie Stevison
  </p>
</div>
<div id="getting-started" class="slide section level2">
<h1>Getting Started</h1>
<p><em>If you are using this code with the video, note that some slides
have been added post-recording and are not shown. See 4.05.walkthrough.R
in the compressed data tarball for the code from the recording.</em></p>
<p>So far we’ve treated R as separate from linux, but you can execute R
from the <a
href="https://en.wikipedia.org/wiki/Command-line_interface">command
line</a>. This is the focus of this tutorial.</p>
<p>Normally we would make an R Notebook, but since we are planning to
use this script on the CL, start an RScript in RStudio. We will work
with a different subset of the depth data from Activity #1.</p>
<p>You will also want to copy the data files into this folder to follow
along, even if you load our data object.</p>
</div>
<div id="set-up-workspace" class="slide section level2">
<h1>Set up workspace</h1>
<ul>
<li><p>You are working within an R Project (check in the top right
corner of RStudio - you should see the project name
“R_Mini_Course”).</p></li>
<li><p>This means that the project directory will also be set as working
directory.</p></li>
<li><p>You should be saving your R Script in the R Project and using
<code>/modules/Module_7/</code> to point at the current module
directory.</p></li>
<li><p>Before starting, it may be helpful to have a chunk of code that
does the following:</p>
<ul>
<li>clear your workspace <code>rm(list=ls())</code></li>
<li>load your packages <code>library(&lt;package-name&gt;)</code></li>
<li>check your session information <code>sessionInfo()</code></li>
<li>list files in your working directory
<code>list.files(getwd())</code></li>
</ul></li>
</ul>
</div>
<div id="read-in-datasets" class="slide section level2">
<h1>Read in datasets</h1>
<p>Let’s use some data from my lab! These are raw data from an
experiment where we measured the impact of temperature on meiotic
recombination rate.</p>
<p>We recorded phenotype as 0’s or 1’s, so we will use R to categorize
the data into crossover classes.</p>
<p>You will need to add path information to the <code>raw_data</code>
directory once you have uncompressed the data tarball.</p>
<p>Read “Exp5_rawdata.csv” into an object called <code>exp5</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#samtools.depth=read.table(file=&quot;chrX.depth.out&quot;)</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#View(samtools.depth)</span></span></code></pre></div>
</div>
<div class="slide section level2">

<p>You may also read in the previously made object:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>samtools.depth <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="at">file =</span> <span class="st">&quot;data/chrX.depth.out.rds&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(samtools.depth, <span class="dv">3</span>)</span></code></pre></div>
<pre><code>##              V1 V2 V3
## 1 chrXL_group3b  1  1
## 2 chrXL_group3b  2  3
## 3 chrXL_group3b  3  4</code></pre>
</div>
<div id="integrating-r-with-the-command-line"
class="slide section level2">
<h1>Integrating R with the command line</h1>
<p>To use R on the command line, we are going to return to a script you
wrote when we first started using R. You could copy the script
<code>coverage.R</code> from Activity #1 to this directory or the code
from it into a new script.</p>
<p>Feel free to clean up the bits in the middle, Remember, we made a
copy - you ALWAYS have the original!</p>
</div>
<div class="slide section level2">

<h3 id="revisiting-our-depth-plot">Revisiting our depth plot</h3>
<p>Let’s recreate the plot we made in Activity #1.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>chrXL_group3b<span class="ot">=</span><span class="fu">subset</span>(samtools.depth,V1<span class="sc">==</span><span class="st">&quot;chrXL_group3b&quot;</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(chrXL_group3b<span class="sc">$</span>V2,chrXL_group3b<span class="sc">$</span>V3,<span class="at">xlab=</span><span class="st">&quot;Position&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Depth of Coverage&quot;</span>,<span class="at">main=</span><span class="st">&quot;chrXL_group3b Coverage&quot;</span>,<span class="at">type=</span><span class="st">&quot;l&quot;</span>,<span class="at">lwd=</span><span class="dv">3</span>,<span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span></code></pre></div>
<p><img src="4.05.R_on_CL_files/figure-slidy/recreate-depth-1.png" width="768" /></p>
</div>
<div class="slide section level2">

<p>That was still a really messy plot. Let’s make some modifications to
improve this figure. First, let’s convert the x axis into megabases.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>mb<span class="ot">=</span>chrXL_group3b<span class="sc">$</span>V2<span class="sc">/</span><span class="dv">10</span><span class="sc">^</span><span class="dv">6</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mb,chrXL_group3b<span class="sc">$</span>V3,<span class="at">xlab=</span><span class="st">&quot;Position (Mb)&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Depth of Coverage&quot;</span>,<span class="at">main=</span><span class="st">&quot;chrXL_group3b Coverage&quot;</span>,<span class="at">type=</span><span class="st">&quot;l&quot;</span>,<span class="at">lwd=</span><span class="dv">3</span>,<span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span></code></pre></div>
<p><img src="4.05.R_on_CL_files/figure-slidy/plot-megabases-1.png" width="768" /></p>
<p>Notice, we’ve change the x in our plot function and the label for the
x-axis to specify that it is in megabases (Mb) for short.</p>
</div>
<div id="other-modifications" class="slide section level2">
<h1>Other Modifications</h1>
<p>Now, let’s add in some bells and whistles that we’ve learned since
Activity #1.</p>
<p>We’ll add an abline for the mean and some text stating the mean as
well.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mb,chrXL_group3b<span class="sc">$</span>V3,<span class="at">xlab=</span><span class="st">&quot;Position (Mb)&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Depth of Coverage&quot;</span>,<span class="at">main=</span><span class="st">&quot;chrXL_group3b Coverage&quot;</span>,<span class="at">type=</span><span class="st">&quot;l&quot;</span>,<span class="at">lwd=</span><span class="dv">3</span>,<span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#add an abline</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>avg<span class="ot">=</span><span class="fu">mean</span>(chrXL_group3b<span class="sc">$</span>V3)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span>avg,<span class="at">col=</span><span class="st">&quot;red&quot;</span>,<span class="at">lwd=</span><span class="dv">3</span>)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co">#add text with mean</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="at">side=</span><span class="dv">3</span>,<span class="fu">paste</span>(<span class="st">&quot;mean = &quot;</span>,<span class="fu">round</span>(avg,<span class="dv">2</span>)),<span class="at">cex=</span><span class="fl">0.8</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>,<span class="at">adj=</span><span class="dv">1</span>)</span></code></pre></div>
<p><img src="4.05.R_on_CL_files/figure-slidy/add-abline-1.png" width="768" /></p>
</div>
<div id="independently-try-again-but-write-to-a-file-this-time."
class="slide section level2">
<h1>Independently, try again but write to a file this time.</h1>
<p>Here, we will write the file to a png instead of a PDF. As some of
you have noticed, a PDF writes the data as vector graphics and thus the
file size is large and takes a long time to load. A png will rasterize
the image, reducing the file size and load time.</p>
<p>Fill in the code to generate the plot where indicated:</p>
<p>Remember, because we wrote this to a file, we need to open the file
to see the image. It will not render in your notebook this time.</p>
</div>
<div
id="play-with-some-of-these-png-paramters-res-heightwidth-and-pointsize-to-see-how-it-changes-the-output-file"
class="slide section level2">
<h1>Play with some of these png paramters (res, height/width, and
pointsize) to see how it changes the output file</h1>
</div>
<div id="smoothing" class="slide section level2">
<h1>Smoothing</h1>
<p>Another thing we can do is smooth the data. One of the better
smoothing functions in R is <code>loess</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>lo<span class="ot">=</span><span class="fu">loess</span>(chrXL_group3b<span class="sc">$</span>V3<span class="sc">~</span>chrXL_group3b<span class="sc">$</span>V2,<span class="at">span=</span><span class="fl">0.01</span>,<span class="fu">data.frame</span>(<span class="at">x=</span>chrXL_group3b<span class="sc">$</span>V2,<span class="at">y=</span>chrXL_group3b<span class="sc">$</span>V3))</span></code></pre></div>
<p>This function will take ~1-2 minutes to run depending on your OS.</p>
</div>
<div class="slide section level2">

<p><code>span=0.01</code> is a 1% smoothing span, which seems very low,
but we have LOTS of data so any value larger would take a VERY long
time.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(chrXL_group3b<span class="sc">$</span>V2)</span></code></pre></div>
<pre><code>## [1] 334023</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(chrXL_group3b<span class="sc">$</span>V2)<span class="sc">*</span><span class="fl">0.01</span></span></code></pre></div>
<pre><code>## [1] 3340.23</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(chrXL_group3b<span class="sc">$</span>V2)<span class="sc">*</span><span class="fl">0.001</span></span></code></pre></div>
<pre><code>## [1] 334.023</code></pre>
</div>
<div class="slide section level2">

<p>We can also smooth using <code>enp.target</code>; This is more useful
if smoothing over different size datasets. It uses a precise number
rather than a percentage.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>lo<span class="ot">=</span><span class="fu">loess</span>(chrXL_group3b<span class="sc">$</span>V3<span class="sc">~</span>chrXL_group3b<span class="sc">$</span>V2,<span class="at">enp.target=</span><span class="fu">tail</span>(chrXL_group3b<span class="sc">$</span>V2,<span class="dv">1</span>)<span class="sc">/</span><span class="dv">500</span>,<span class="fu">data.frame</span>(<span class="at">x=</span>chrXL_group3b<span class="sc">$</span>V2,<span class="at">y=</span>chrXL_group3b<span class="sc">$</span>V3))</span></code></pre></div>
</div>
<div id="what-does-the-enp.target-look-like"
class="slide section level2">
<h1>What does the enp.target look like?</h1>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(chrXL_group3b<span class="sc">$</span>V2,<span class="dv">1</span>)</span></code></pre></div>
<pre><code>## [1] 386180</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(chrXL_group3b<span class="sc">$</span>V2,<span class="dv">1</span>)<span class="sc">/</span><span class="dv">500</span></span></code></pre></div>
<pre><code>## [1] 772.36</code></pre>
</div>
<div id="span-versus-enp.target" class="slide section level2">
<h1><code>span</code> versus <code>enp.target</code></h1>
<p>Here, we will compare how <code>span</code> and
<code>enp.target</code> differ when you are comparing two chromosomes.
As we’ve done before, we will start with a blank plot and add each piece
in one at a time.</p>
<p>We are making the same plot, but using a larger smoothing span</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>lo1<span class="ot">=</span><span class="fu">loess</span>(chrXL_group3b<span class="sc">$</span>V3<span class="sc">~</span>chrXL_group3b<span class="sc">$</span>V2,<span class="at">span=</span><span class="fl">0.001</span>,<span class="fu">data.frame</span>(<span class="at">x=</span>chrXL_group3b<span class="sc">$</span>V2,<span class="at">y=</span>chrXL_group3b<span class="sc">$</span>V3))</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mb,chrXL_group3b<span class="sc">$</span>V3,<span class="at">ylim=</span><span class="fu">range</span>(<span class="fu">predict</span>(lo1)),<span class="at">main=</span><span class="st">&quot;chrXL_group3b Coverage&quot;</span>,<span class="at">xlab=</span><span class="st">&quot;Position (Mb)&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Coverage&quot;</span>,<span class="at">type=</span><span class="st">&quot;n&quot;</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="co">#add loess prediction</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(mb,<span class="fu">predict</span>(lo1),<span class="at">lwd=</span><span class="dv">3</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="4.05.R_on_CL_files/figure-slidy/span2-1.png" width="768" /></p>
</div>
<div class="slide section level2">

<p>Now, try to repeat the process with the other chromosome. Go back to
your code and subset for chrXR_group3a</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#chrXR_group3a= </span></span></code></pre></div>
<p>Continue to the next slide for the solution.</p>
</div>
<div class="slide section level2">

<p>Solution:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>chrXR_group3a<span class="ot">=</span><span class="fu">subset</span>(samtools.depth,samtools.depth<span class="sc">$</span>V1<span class="sc">==</span><span class="st">&quot;chrXR_group3a&quot;</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>mb2<span class="ot">=</span>chrXR_group3a<span class="sc">$</span>V2<span class="sc">/</span><span class="dv">10</span><span class="sc">^</span><span class="dv">6</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Perform loess for a much longer chromosome:</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>lo2<span class="ot">=</span><span class="fu">loess</span>(chrXR_group3a<span class="sc">$</span>V3<span class="sc">~</span>chrXR_group3a<span class="sc">$</span>V2,<span class="at">span=</span><span class="fl">0.001</span>,<span class="fu">data.frame</span>(<span class="at">x=</span>chrXR_group3a<span class="sc">$</span>V2,<span class="at">y=</span>chrXR_group3a<span class="sc">$</span>V3))</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mb2,chrXR_group3a<span class="sc">$</span>V3,<span class="at">ylim=</span><span class="fu">range</span>(<span class="fu">predict</span>(lo2)),<span class="at">main=</span><span class="st">&quot;chrXR_group3a Coverage&quot;</span>,<span class="at">xlab=</span><span class="st">&quot;Position (Mb)&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Coverage&quot;</span>,<span class="at">type=</span><span class="st">&quot;n&quot;</span>)</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="co">#add loess prediction</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(mb2,<span class="fu">predict</span>(lo2),<span class="at">lwd=</span><span class="dv">3</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="4.05.R_on_CL_files/figure-slidy/span-solution-1.png" width="768" /></p>
</div>
<div class="slide section level2">

<p>Let’s compare the range of data with and without smoothing:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co">#range of data without smoothing</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;chrXL_group3b</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## chrXL_group3b</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(chrXL_group3b<span class="sc">$</span>V3)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##    0.00    8.00   11.00   11.82   14.00   71.00</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;chrXL_group3a</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## chrXL_group3a</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(chrXR_group3a<span class="sc">$</span>V3)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##    0.00    7.00   10.00   10.44   13.00   70.00</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co">#range of smoothed data</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;chrXL_group3b</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## chrXL_group3b</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">predict</span>(lo1))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.1497  8.7432 10.7693 11.8515 12.9247 64.6999</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;chrXL_group3a</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## chrXL_group3a</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">predict</span>(lo2))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   1.939   8.643  10.173  10.447  11.558  55.399</code></pre>
</div>
<div class="slide section level2">

<p>Using span, we are smoothing over a different number of parameters
for the two datasets based on their size. This makes it hard to compare
from chromosome to chromosome. <code>enp.target</code> is preferred
method for chromosomes so they can be compared.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>lo3<span class="ot">=</span><span class="fu">loess</span>(chrXR_group3a<span class="sc">$</span>V3<span class="sc">~</span>chrXR_group3a<span class="sc">$</span>V2,</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">enp.target =</span> <span class="fu">tail</span>(chrXR_group3a<span class="sc">$</span>V2,<span class="dv">1</span>)<span class="sc">/</span><span class="dv">500</span>,</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>          <span class="fu">data.frame</span>(<span class="at">x=</span>chrXR_group3a<span class="sc">$</span>V2,<span class="at">y=</span>chrXR_group3a<span class="sc">$</span>V3))</span></code></pre></div>
</div>
<div class="slide section level2">

<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co">#range of data without smoothing</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;chrXL_group3b</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## chrXL_group3b</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(chrXL_group3b<span class="sc">$</span>V3)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##    0.00    8.00   11.00   11.82   14.00   71.00</code></pre>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;chrXL_group3a</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## chrXL_group3a</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(chrXR_group3a<span class="sc">$</span>V3)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##    0.00    7.00   10.00   10.44   13.00   70.00</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co">#range of smoothed data</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;chrXL_group3b</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## chrXL_group3b</code></pre>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">predict</span>(lo))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   3.258   9.495  10.790  11.828  12.158  54.154</code></pre>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;chrXL_group3a</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## chrXL_group3a</code></pre>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">predict</span>(lo3))</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   2.626   8.729  10.169  10.448  11.472  52.104</code></pre>
</div>
<div id="integrate-smoothed-data-with-previous-plot"
class="slide section level2">
<h1>Integrate smoothed data with previous plot</h1>
<p>Let’s add this smoothed data on top of the raw data. Make the
appropriate modifications to the code to write the plot to a file.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Let&#39;s see our new plot.</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mb,chrXL_group3b<span class="sc">$</span>V3,<span class="at">main=</span><span class="st">&quot;chrXL_group3b Coverage&quot;</span>,<span class="at">xlab=</span><span class="st">&quot;Position (Mb)&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;Coverage&quot;</span>,<span class="at">type=</span><span class="st">&quot;l&quot;</span>,<span class="at">col=</span><span class="st">&quot;grey50&quot;</span>)</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a><span class="co">#only text of mean needed, not abline, because smoothed function is better!</span></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="at">side=</span><span class="dv">3</span>,<span class="fu">paste</span>(<span class="st">&quot;mean = &quot;</span>,<span class="fu">round</span>(avg,<span class="dv">2</span>)),<span class="at">cex=</span><span class="fl">0.8</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>,<span class="at">adj=</span><span class="dv">1</span>)</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(mb,<span class="fu">predict</span>(lo),<span class="at">lwd=</span><span class="dv">2</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="4.05.R_on_CL_files/figure-slidy/unnamed-chunk-8-1.png" width="768" /></p>
</div>
<div id="scaling-up" class="slide section level2">
<h1>Scaling up:</h1>
<p>This is a nice script, but this is only the coverage along a single
contig for our dataset.</p>
<p>We have 13 contigs and would like to see how coverage looks across
ALL contigs. This is a bit much for our local machine to do.</p>
<p>To run this R script on the ASC, we will need to transfer it up to
the machine and write a shell script to change the name of the
chromosome each time.</p>
</div>
<div class="slide section level2">

<p>In Activity #3 you can practice using R on the CL You will execute
the script for 1 contig only first to test the script works on an
HPC.</p>
<p>An easy way to execute an R script on the command line:</p>
<p><code>R --vanilla &lt;coverage.R</code></p>
<p>If you decide not to perform Activity #3 or you do not have access to
HPC:</p>
<p>Congratulations, you’ve completed this mini course!</p>
</div>

  <!-- dynamically load mathjax for compatibility with self-contained -->
  <script>
    (function () {
      var script = document.createElement("script");
      script.type = "text/javascript";
      script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
      document.getElementsByTagName("head")[0].appendChild(script);
    })();
  </script>

</body>
</html>
